### ISSUE TITLE
For new datasets

### ISSUE THREAD

**Issue by jindongwang** [qqwweee/keras-yolo3#30]
I want to use the code to detect the ID card. Right now I have some data collected and labeled. Is it OK I run train.py with only my dataset, which only has 1 class: id card?

**Comment by qqwweee** [comment]
It seems OK in theory, but the effect is hard to say.
Just try and see the result, then maybe improve training configure.

**Comment by jindongwang** [comment]
@qqwweee Thank you! I'm training now.

**Comment by jindongwang** [comment]
@qqwweee I tried with my own datasets: ID card images with labelled bounding box. The goal is to idenfity the card region itself. So there is only 1 class: ID card. I tried with the pretrained and non-pretrained setting, but the pretrained version only gives a error, which is not the results (predicted bounding box).
The non-pretrained version gives an erros like:
![qq 20180507134302](https://user-images.githubusercontent.com/6158206/39686702-37a5edf6-51fd-11e8-8ef7-5bb3b1867442.png)

**Comment by qqwweee** [comment]
nan is caused by numerical problem, like overflow or underflow.
For your pretrained version, what is the final loss? Consider if it is necessary to train a bit longer.

**Comment by jindongwang** [comment]
The pretrained version works normal with the final loss to be around 2. But I can't see the result (picture with bounding box).
What should I do with the non-pretrained version?

**Comment by qqwweee** [comment]
Non-pretrained model is deprecated.
The weight L2 item in total loss is ~10, so your final loss cannot be 2, I think.
Actually, I didn't do a lot of try on training, so maybe cannot help you much.

**Comment by jindongwang** [comment]
Ok, thank you anyway! I will keep trying!

**Comment by zhouzhubin** [comment]
@jindongwang 你好，我想请问你在训练的过程中出现过 MemoryError吗？

**Comment by jindongwang** [comment]
@zhouzhubin 有过

**Comment by zhouzhubin** [comment]
@jindongwang 你是怎么解决的的？我用23000多张图片做训练，生成的npz文件有11G

**Comment by jindongwang** [comment]
@zhouzhubin 我暂时没解决

**Comment by nstp** [comment]
@jindongwang 你的nan问题解决了吗？

**Comment by Amerry-c** [comment]
@jindongwang @qqwweee @zhouzhubin How is the .npz file obtained? It's strange that I was able to get the npz file a few days ago, but it is not now. I don't know what I missed

**Comment by MarcHo1994** [comment]
@zhouzhubin MemoryError 在Keras中就是内存错误，npz文件有11G,太大了，不要把23000张数据一起做成npz,试试将数据转成两个npz,先训练一个，然后再训练另一个。简单说，就是不要一次性把所有数据都读入内存训练，电脑内存没那么大啊！

**Comment by wizholy** [comment]
@jindongwang 你好，如何用新数据自定义的classes文件重新训练，同时使用yolo.h5进行微调。代码可以分享一下吗？谢谢
